src/parameter_bridge.cpp

```diff
@@ -86,9 +86,15 @@ int main(int argc, char * argv[])
         "with ROS 2 type '%s'\n",
         topic_name.c_str(), type_name.c_str());
 
+      auto ros2_publisher_qos = rclcpp::QoS(rclcpp::KeepLast(queue_size));
+      if (topic_name == "/tf_static") {
+        ros2_publisher_qos.keep_all();
```

Comment:
I don't think this should set `keep_all`. Upstream uses a queue depth of 1 for broadcasters, or 100 for listeners. It seems reasonable to keep using `KeepLast(queue_size)`.

https://github.com/ros2/geometry2/blob/4a6526b3e861f22639460750cd197b59ac1a100b/tf2_ros/include/tf2_ros/qos.hpp#L56

Reply:
I think the bridge's ROS 2 subscriber would need to use `transient_local()` too, otherwise I think it won't get any old samples. Similarly the ROS 1 publisher would need to use `latching`.

Reply:
Hi there,
the settings were adopted from those used in #282 so they are equal to the dynamic bridge.
I assumed the history policy was set to `keep_all` as the messages on the `tf_static` topic are often only published once as far as I know so any late-joining subscriber would need to get all the messages transported on the topic. But I could be wrong here. If I change the setting, should I also change it in the dynamic bridge?

If I understand the QoS documentation (https://docs.ros.org/en/foxy/Concepts/About-Quality-of-Service-Settings.html) correctly I think changing the ROS2 subscriber to use `transient_local` would have the effect that the sub would only accept messages from publishers that are publishing with a `transient_local` policy, too. It would ignore `volatile` publishers. So in order for the sub to receive old samples, I don't think it needs to use `transient_local`. Instead the ROS2 publishers that publish the static TFs would need to be `transient_local`. Correct me if I'm mistaken here, the QoS settings are still pretty new to me.

I agree that the ROS1 publisher would need to use `latching`. This is also not done in the dynamic bridge at the moment. I added a commit that activates ROS1 pub latching if the topic to bridge is `tf_static` (for both the dynamic & static bridge).